tests/integration/test_application.py

```diff
@@ -147,26 +137,24 @@ async def test_deploy_charmhub_charm(event_loop):
 @pytest.mark.asyncio
 async def test_upgrade_charm(event_loop):
     async with base.CleanModel() as model:
-        app = await model.deploy('cs:ubuntu-0')
+        app = await model.deploy('ubuntu')
         await model.block_until(lambda: (len(app.units) > 0 and
                                          app.units[0].machine))
-        assert app.data['charm-url'] == 'cs:ubuntu-0'
+        assert app.data['charm-url'] == 'ubuntu'
         await app.upgrade_charm()
-        assert app.data['charm-url'].startswith('cs:ubuntu-')
-        assert app.data['charm-url'] != 'cs:ubuntu-0'
+        assert app.data['charm-url'] != 'ubuntu'
 
 
 @base.bootstrapped
 @pytest.mark.asyncio
 async def test_upgrade_charm_channel(event_loop):
     async with base.CleanModel() as model:
-        app = await model.deploy('cs:ubuntu-0')
+        app = await model.deploy('ubuntu')
         await model.block_until(lambda: (len(app.units) > 0 and
                                          app.units[0].machine))
-        assert app.data['charm-url'] == 'cs:ubuntu-0'
+        assert app.data['charm-url'] == 'ubuntu'
         await app.upgrade_charm(channel='stable')
```

Comment:
same in this one, the deploy will default to the `stable` risk, so this test wouldn't be upgrading anything.

Reply:
I will simply leave `test_upgrade_charm_switch_channel` which has all the logic for this kind of testing.